packages/flutter_tools/lib/src/device.dart

```diff
@@ -915,6 +915,42 @@ class DebuggingOptions {
 
   bool get hasObservatoryPort => hostVmServicePort != null;
 
+  List<String> getIOSLaunchArguments(EnvironmentType environmentType, String? route,  Map<String, Object?> platformArgs) {
+    final String dartVmFlags = computeDartVmFlags(this);
+    return <String>[
+      '--enable-dart-profiling',
+      if (environmentType == EnvironmentType.physical) '--disable-service-auth-codes',
+      if (environmentType == EnvironmentType.simulator && disableServiceAuthCodes) '--disable-service-auth-codes',
+      if (environmentType == EnvironmentType.physical && disablePortPublication) '--disable-observatory-publication',
```

Comment:
This can also be passed into the simulator, if it's set.  For some reason I thought we didn't do mDNS publication for simulator, but we do:
https://github.com/flutter/engine/blob/a339093f6593e4e155451019af06f333d8f3e325/shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm#L79-L81
https://github.com/flutter/flutter/blob/be83c98edda77ee245764c264752fadf8f2cb8fa/packages/flutter_tools/lib/src/commands/attach.dart#L305-L307
```suggestion
      if (disablePortPublication) '--disable-observatory-publication',
```

Reply:
Changed in https://github.com/flutter/flutter/pull/114628/commits/c6b500e5dc835ccba5041ca51119424cba078c0c.